Task 2 - Task 02 - Project infrastructure

nikolay-borzov opened the pull request:
resolves #3

Most of the changes are already in [master](https://github.com/nikolay-borzov/react-mentoring/commit/2736f793ad43cc21e14155e8117842ad6613ce92)

This PR 
- updates ESLint config
- configures Jest
- adds bundle analyzer
- rewrites `server.js` to build and run both dev and prod bundles

Commit:
chore: Configure Heroku

Commit:
Change pocfile

Commit:
chore: change Pocfile

Commit:
Procfile

Commit:
chore: Build for production using Webpack Node.js API

- Wire up `webpack-bundle-analyzer`

nikolay-borzov committed:
Merge branch 'master' into heroku-play

nikolay-borzov committed:
chore: Configure `jest`

- Use `eslint-config-prettier-standard`

Commit:
chore: Precommit and prepush hooks

tsvetkovv commented on webpack.common.js:
```diff
@@ -36,37 +31,53 @@ module.exports = {
-      }
-    ]
+    output: {
+      filename: '[name].bundle.js',
```
Good practice to have a hash of asset to invalidate a browser cache when a bundle is changed
[See here](https://survivejs.com/webpack/optimizing/adding-hashes-to-filenames/) for details

nikolay-borzov replied:
I'll add it

Commit:
chore (task_02): Configure webpack

- Several changes after visiting a webpack training

Commit:
chore (task_02): Minimize CSS

- Move wiring up `ExtractTextPlugin` to common for simplicity's sake

nikolay-borzov committed:
chore(webpack):  Add hash to files URLs

- Move CleanWebpackPlugin to common config

nikolay-borzov committed:
chore(webpack): Prod config improvements

- Add hasg to file URL only for prod builds
- Use ExtractTextPlugin only for prod builds
